### ISSUE TITLE
show method errors with view with non standard indices

### ISSUE THREAD

**Issue by piever** [JuliaLang/julia#31641]
On Julia-1.1 (but master seems to have the same issue) one can take a `view` with offset indices. That returns the correct result, but the show method (with `:limit => true`) errors:

```julia
julia> using OffsetArrays

julia> s = rand(1000);

julia> i = OffsetArray(400:900, 400:900);

julia> axes(i)
(Base.IdentityUnitRange(400:900),)

julia> view(s, i)
view(::Array{Float64,1}, Error showing value of type SubArray{Float64,1,Array{Float64,1},Tuple{OffsetArray{Int64,1,UnitRange{Int64}}},false}:
ERROR: BoundsError: attempt to access Base.IdentityUnitRange{UnitRange{Int64}} with indices 400:900 at index [1]
```

**Comment by timholy** [comment]
Thanks for the report. Triggered only under conditions where `length(v) > 20`, `:limit=>true`, `ndims==1`. Fix in #31642.

**Comment by piever** [comment]
Thank you for the quick fix!
